File: src/main/java/com/bedirhanatasoy/game/tradingcard/player/Player.java
package com.bedirhanatasoy.game.tradingcard.player;

import lombok.Getter;

import java.util.LinkedList;
import java.util.List;
import java.util.stream.IntStream;

@Getter
public class Player {

    private String name;

    private int health;

    private int mana;

    private int manaSlot;

    private Deck deck;

    private List<Card> cards;

    public Player(String name, int health, int manaSlot, Deck deck) {
        this.name = name;
        this.health = health;
        this.manaSlot = manaSlot;
        this.deck = deck;
        this.cards = new LinkedList<>();
    }

    public Card useCard(int cardIndex) {
        if (cardIndex >= cards.size()) {
            return null;
        }
        Card card = cards.get(cardIndex);
        if (card.getManaCost() > mana) {
            return null;
        }
        decreaseMana(card.getManaCost());
        return cards.remove(cardIndex);
    }

    public void decreaseHealth(int amount) {
        health -= amount;
    }

    public void decreaseMana(int amount) {
        mana -= amount;
    }

    public void increaseManaSlot(int amount) {
        manaSlot += amount;
    }

    public void drawRandomCardsFromDeck(int count) {
        IntStream.range(0, count).forEach(i -> cards.add(deck.getRandomCard()));
    }

    public void setMana(int mana) {
        this.mana = mana;
    }

}

File: src/main/kotlin/com/bedirhanatasoy/game/card/player/Player.kt
package com.bedirhanatasoy.game.card.player

data class Player(
    var name: String,
    var health: Int,
    var manaSlot: Int,
    val deck: Deck,
    val cards: MutableList<Card> = mutableListOf()
) {

    var mana: Int = 0

    fun useCard(cardIndex: Int): Card? =
        cards[cardIndex]
            .takeIf { it.manaCost <= mana }
            ?.also { cards.remove(it) }
            ?.also { decreaseMana(it.manaCost) }

    fun increaseManaSlot(amount: Int) {
        manaSlot += amount
    }

    fun drawRandomCardsFromDeck(amount: Int) {
        repeat(amount) { cards.add(deck.getRandomCard()) }
    }

    fun decreaseHealth(amount: Int) {
        health -= amount
    }

    private fun decreaseMana(amount: Int) {
        mana -= amount
    }

}